### ISSUE TITLE
Error: Rounded quality exceeded range of err lookup table.

### ISSUE THREAD

**Issue by shokoiwai** [benjjneb/dada2#59]
Hi,
I got following error while running DADA2 (version ‘0.10.2’). How can I resolve this error? Thanks!

Attempting DADA2: 
_sample_name_.fastq.gz
Sample 1 - 160605 reads in 40125 unique sequences.
   selfConsist step 2 
Error: Error: Rounded quality exceeded range of err lookup table.

**Comment by benjjneb** [comment]
I have never seen that error get triggered before...

Could you provide the full call you are making to the dada(...) function?
Also, could you run the following commands and report the result:

```
min(your.derep.object$quals)
max(your.derep.object$quals)
````

**Comment by shokoiwai** [comment]
Thank you for the quick response. Function I ran was,

dadaRes <- dada(derep,
             err = inflateErr(tperr1, 3), 
             errorEstimationFunction = loessErrfun, 
             selfConsist = TRUE)

min(your.derep.object$quals)
12
max(your.derep.object$quals)
39.5

**Comment by benjjneb** [comment]
Hm.... Can you try one more command and post the output?

```
table(nchar(getSequences(your.derep.object)))
```

**Comment by shokoiwai** [comment]
---
  195 
40125
---

**Comment by shokoiwai** [comment]
195 
40125

**Comment by benjjneb** [comment]
I'm at a bit of a loss. Is it possible for you to share the derep-class object that is causing this behavior? If I can replicate the error I can figure out what's going on.

Ideal would be an .rds file of the derep-object, i.e.

```
saveRDS(derep, "rounding_bug.rds")
```

And share the rounding_bug.rds file.

**Comment by benjjneb** [comment]
Fixed in f1ce503ce7c764a06e736762477b586b8574b0e7

Was a bug where qmax was being rounded down when passed from R-side (as a double) to C-side (cast to unsigned int).

**Comment by shokoiwai** [comment]
Thank you!!
